### PR TITLE
onExit unit test

### PR DESCRIPTION
**Pull request by sandeepcongh**
onExit unit test

### CODE REVIEW

File: js/tests/launchAsync.test.ts [hunk 1]
```diff
@@ -217,6 +217,31 @@ describe('launchAsync tests', () => {
...
+
+        const options: Options = { onExit: () => { cbOnExit(); } };
+        const launchPromise = launchAsync(SampleToken, null, SampleContent, options);
+        window.postMessage('ImmersiveReader-LaunchSuccessful', '*');
+
+        await launchPromise;
+
+        window.postMessage('ImmersiveReader-Exit', '*');
+
+        const timeout = (ms: number) => new Promise(resolve => { setTimeout(resolve, ms); });
```
Comment (rwallerms) [inline js/tests/launchAsync.test.ts]: can you condense the timeout code to something like the following?

    await new Promise(resolve => { setTimeout(resolve, 1); });

and add comments that this is to yield this thread of execution to allow the exit message to get processed

Comment (sandeepcongh) [reply js/tests/launchAsync.test.ts]: i have condense the timeout code and added the comment

File: js/tests/launchAsync.test.ts [hunk 2]
```diff
@@ -217,6 +217,26 @@ describe('launchAsync tests', () => {
...
+        const options: Options = { onExit: () => { cbOnExit(); } };
+        const launchPromise = launchAsync(SampleToken, null, SampleContent, options);
+        window.postMessage('ImmersiveReader-LaunchSuccessful', '*');
+
+        await launchPromise;
+
+        window.postMessage('ImmersiveReader-Exit', '*');
+
+        // this is to yield this thread of execution to allow the exit message to get processed
+        await new Promise(resolve => { setTimeout(resolve, 500); });
```
Comment (rwallerms) [inline js/tests/launchAsync.test.ts]: change this to 1 ms

File: js/tests/launchAsync.test.ts [hunk 3]
```diff
@@ -217,6 +217,31 @@ describe('launchAsync tests', () => {
...
+
+        const timeout = (ms: number) => new Promise(resolve => { setTimeout(resolve, ms); });
+
+        const waitFor500MilliSeconds = async () => {
+          await timeout(500);
+        };
+        
+        await waitFor500MilliSeconds();
+
+        expect(cbOnExit.mock.calls.length).toBe(1);
```
[the remark this answers is not in the document]
Comment (rwallerms) [reply js/tests/launchAsync.test.ts]: or this - https://jestjs.io/docs/en/expect#tohavebeencalledtimesnumber

[the remark this answers is not in the document]
Comment (sandeepcongh) [reply js/tests/launchAsync.test.ts]: i have changed it to use 'toHaveBeenCalled' and set the wait time to 1 milli second

**Pull request merged**